File: sub_and_pub_template.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include "std_msgs/Float64.h"


class SubscribeAndPublish 
{
private:
    // ROS
    ros::NodeHandle n_; 
    ros::Subscriber twist_sub;
    ros::Publisher float_pub;

public:
    SubscribeAndPublish() 
    {
        // Subscriber
        twist_sub = n_.subscribe("pitch_yaw", 10, &SubscribeAndPublish::process_angle, this);
        // Publisher
        float_pub = n_.advertise<std_msgs::Float64>("angle", 1);
    }
    ~SubscribeAndPublish() {}

    void process_angle(const geometry_msgs::Twist twist)
    {
        // Initialize published message
        std_msgs::Float64 angle;

        // Map to angle
        float angle_map = 1.047;
        angle.data = twist.angular.z * angle_map;

        // ROS_DEBUG("Get angle %f", angle.data);
        float_pub.publish(angle);
    }
};


int main(int argc, char *argv[])
{
    // Launch ros node class
    ros::init(argc, argv, "sub_and_pub_template_node");
    SubscribeAndPublish SAP;

    // Set rate and run
    ros::Rate rate(60);
    while (ros::ok())
        ros::spinOnce();
        rate.sleep();
    return 0;
}

File: sub_and_pub_template.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String
from geometry_msgs.msg import Twist


class SubAndPub():

    def __init__(self):
        # ROS
        # subscribe to keyboard input
        self.subsciber = rospy.Subscriber("key", String, self.get_key_callback)
        # publish the control command
        self.publisher = rospy.Publisher("cmd_vel", Twist, queue_size=1)


    def get_key_callback(self, data):
        # Initialize published message
        twist = Twist()
        
        # Map to command
        key = data.data
        if key == "w":
            twist.linear.x = 1.0
        if key == "s":
            twist.linear.x = -1.0
        if key == "a":
            twist.angular.z = 1.0
        if key == "d":
            twist.angular.z = -1.0

        # rospy.loginfo("Send twist commad: Lx: ", twist.linear.x, ", Az: ", twist.angular.z)
        self.publisher.publish(twist)


if __name__ == "__main__":
    # Launch ros node class
    rospy.init_node('sub_and_pub_template_node')
    sub_and_pub = SubAndPub()

    # Set rate and run
    r = rospy.Rate(60)
    while not rospy.is_shutdown():
        r.sleep()
